src/main/java/schwimmer/mco364/paint/PencilTool.java

```diff
@@ -12,25 +15,27 @@ public PencilTool(PaintProperties properties) {
 		super(properties);
 	}
 
-	public void mousePressed(Graphics g, int x, int y) {
+	public void mousePressed(Graphics2D g, int x, int y) {
 		g.setColor(properties.getColor());
+		g.setStroke(new BasicStroke(properties.getWeight()));
```

Comment:
Don't create a new BasicStroke object every single time the mouse is pressed. Move BasicStroke into PaintProperties and change this also on line 32.